sycl/doc/extensions/SubGroupNDRange/SubGroupNDRange.md

```diff
@@ -1,286 +1,7 @@
-
-### Loads / Stores
-
-The load and store sub-group functions enable developers to assert that all work-items in a sub-group read/write from/to contiguous locations in memory.  Such operations can be mapped directly to SIMD operations.
-
-|Member functions|Description|
-|----------------|-----------|
-| `template <typename T, access::address_space Space>T load(const multi_ptr<T,Space> src) const` | Load contiguous data from `src`.  Returns one element per work-item, corresponding to the memory location at `src` + `get_local_id()`. The value of `src` must be the same for all work-items in the sub-group. |
-| `template <int N, typename T, access::address_space Space>vec<T,N> load(const multi_ptr<T,Space> src) const` | Load contiguous data from `src`.  Returns `N` elements per work-item, corresponding to the `N` memory locations at `src` + `i` * `get_max_local_range()` + `get_local_id()` for `i` between 0 and `N`. The value of `src` must be the same for all work-items in the sub-group. |
-| `template <typename T, access::address_space Space>void store(multi_ptr<T,Space> dst, const T& x) const` | Store contiguous data to `dst`.  The value of `x` from each work-item is written to the memory location at `dst` + `get_local_id()`. The value of `dst` must be the same for all work-items in the sub-group. |
-| `template <int N, typename T, access::address_space Space>void store(multi_ptr<T,Space> dst, const vec<T,N>& x) const` | Store contiguous data to `dst`.  The `N` elements from each work-item are written to the memory locations at `dst` + `i` * `get_max_local_range()` + `get_local_id()` for `i` between 0 and `N`.  The value of `dst` must be the same for all work-items in the sub-group. |
-
-# Sample Header
-
-```c++
-namespace cl {
-namespace sycl {
-namespace intel {
-struct sub_group {
-
-    /* --- common interface members --- */
-
-    id<1> get_local_id() const;
-
-    range<1> get_local_range() const;
-
-    range<1> get_max_local_range() const;
-
-    id<1> get_group_id() const;
-
-    uint32_t get_group_range() const;
-
-    uint32_t get_uniform_group_range() const;
-
-    /* --- vote/ballot functions --- */
-
-    bool any(bool predicate) const;
-
-    bool all(bool predicate) const;
-
-    /* --- data-sharing --- */
-
-    template <typename T>
-    T broadcast(T x, id<1> local_id) const;
-
-    template <typename T, class BinaryOp>
-    T reduce(T x, BinaryOp binary_op) const;
-
-    template <typename T, class BinaryOp>
-    T reduce(T x, T init, BinaryOp binary_op) const;
-
-    template <typename T, class BinaryOp>
-    T exclusive_scan(T x, BinaryOp binary_op) const;
-
-    template <typename T, class BinaryOp>
-    T exclusive_scan(T x, T init, BinaryOp binary_op) const;
-
-    template <typename T, class BinaryOp>
-    T inclusive_scan(T x, BinaryOp binary_op) const;
-
-    template <typename T, class BinaryOp>
-    T inclusive_scan(T x, BinaryOp binary_op, T init) const;
```

Comment:
I'm not sure where is the best place to ask this, but `inclusive_scan` seems inconsistent with `exclusive_scan` and `reduce` - `init` is the second argument in the other two functions, but the third argument here. This is also true of the document here: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/GroupAlgorithms/SYCL_INTEL_group_algorithms.asciidoc

Reply:
The inconsistency here is unfortunate, but we inherited it from standard C++.  See [`std::reduce`](https://en.cppreference.com/w/cpp/algorithm/reduce), [`std::exclusive_scan`](https://en.cppreference.com/w/cpp/algorithm/exclusive_scan) and [`std::inclusive_scan`](https://en.cppreference.com/w/cpp/algorithm/inclusive_scan).

My understanding is that this ordering is aligned with the ability to provide defaults for each argument: an inclusive scan is well defined without an initial value, but an exclusive scan is not.

Reply:
Thanks, I wasn't aware of that

Reply:
It is more for a mathematical reason than a C++ inconsistency.

For example for `+` the neutral element is 0 but for `*` the neutral element is 1. Obviously for all other data types and other operations it it is different, not even talking about non-mathematical use of these prefix operations, such as implementing a parser with scans...